polynote-kernel/src/main/scala/polynote/kernel/util/TaskQueue.scala

```diff
@@ -99,13 +97,14 @@ class TaskQueue(
         _       = waitingTasks.remove(id)
         done    = ReadySignal()
         join    = fiber.join.guarantee(currentTaskRef.set(None)).guarantee(done.complete)
-        update <- statusUpdates.subscribe(32)
-          .collect { case UpdatedTasks(tasks) => tasks.filter(ti => ti.id == id && ti.status != TaskStatus.Complete) }
-          .flatMap(Stream.emits)
-          .evalMap[IO, Unit](updated => currentTaskRef.update(opt => opt.filter(_.taskInfo.id == id).map(_.copy(taskInfo = updated)).orElse(opt)))
-          .interruptWhen[IO](done()).compile.drain.start
+// TODO: another way of doing the below
```

Comment:
So was `update` essentially unused? It looks like it's side-effect updating `currentTaskRef` if I'm reading it right, is that no longer needed?

If so, any reason to keep the code around?

Reply:
The only reason I commented it was to remind me to have a way of accomplishing this – I'll delete it and replace with an explanatory comment instead.

Here's what it was doing: We have a task, which we've notified everyone is running. Now, the task might also be sending progress updates, and `TaskQueue` has a way of getting the current task (i.e. so if you reload you can see it). But if the progress is being updated out-of-band, that current task won't be up-to-date. So it just listens for any other updates to the task and updates itself. Pretty silly and roundabout way of doing it, and since I removed the ability to subscribe the status updates here, it can't be done that way anymore.

Hopefully centralizing all the task updates would solve this problem.

Reply:
Ah, gotcha. I read the TODO as "(here's) another way of doing the below" whereas you meant, literally, "TODO: another way of doing the below" 🤦‍♂️No idea why I read it the other way which doesn't actually make sense... 

On the other hand, your explanation helped me understand what the code was doing and was really helpful!

Reply:
Removed this dead code